lib/gingr/solr_indexer.rb

```diff
@@ -1,49 +1,44 @@
 # frozen_string_literal: true
 require 'faraday/net_http_persistent'
+require 'find'
 require 'rsolr'
 require_relative 'config'
 
 module Gingr
   class SolrIndexer
     include Config
 
-    attr_reader :reference_urls
-    attr_reader :solr
+    attr_accessor :reference_urls
+    attr_accessor :solr
 
-    def initialize(url, reference_urls = {})
-      @solr = RSolr.connect url:, adapter: :net_http_persistent
-      @reference_urls = reference_urls
+    def initialize(solr = nil, reference_urls = nil)
+      solr ||= ENV['SOLR_URL'] || Gingr::Config.default_options[:solr_url]
+      solr = RSolr.connect url: solr, adapter: :net_http_persistent if solr.kind_of? String
+      @solr = solr
+      @reference_urls = reference_urls || {}
     end
 
-    def update_reference_urls?
-      !@reference_urls.empty?
+    def add(doc)
+      doc = JSON.load_file(doc) if doc.kind_of? String
+      update_reference_urls!(doc)
+      @solr.add doc
     end
 
-    def update(file_path)
-      commit_within = ENV.fetch('SOLR_COMMIT_WITHIN', 5000).to_i
-      doc = JSON.parse(File.read(file_path))
-      [doc].flatten.each do |record|
-        update_reference_urls!(record) if update_reference_urls?
-        @solr.update params: { commitWithin: commit_within, overwrite: true },
-                     data: [record].to_json,
-                     headers: { 'Content-Type' => 'application/json' }
-      end
-    end
-
-    def commit
-      @solr.commit
+    def index_directory(directory)
+      Find.find(directory)
+        .select(&method(:json_file?))
+        .each(&method(:add))
```

Comment:
It made sense to consolidate the parsing/indexing logic into the indexer rather than split it between here and ImportUtils. The method becomes considerably more concise as a result.

Reply:
Yes, thanks

Reply:
Yes, it is more concise